lib/account-handler.js

```diff
@@ -1,133 +1,109 @@
-    }
-}
+const {checkNickname,removeAccount,switchAccountData,getAccountList } = require('./account-data')
+const {error} = require("./error");
+const pWaitFor  = require('p-wait-for')
+
+
 async function stopParsec(){
-    const output = await findProcess('parsecd')
-    const op2 = await findProcess('pservice')
-    //console.log(JSON.stringify(op2))
-    //console.log(JSON.stringify(output))
+    //todo logger
 
 
-    if (op2.success){
+    if (await findProcess('pservice')){
         const stopService = await stopProcess("pservice.exe")
-        if (!stopService.success){
-            throw new Error(stop.message)
+        if (!stopService){
+            return error.STOP_FAILED
+            //todo logger
         }
     }
-    if(output.success){
+    if(await findProcess('parsecd')){
         const stop = await stopProcess("parsecd.exe")
-        if (!stop.success){
-            throw new Error(stop.message)
+        if (!stop){
+            return error.STOP_FAILED
+            //todo logger
         }
-        //console.log(JSON.stringify(stop))
     }
 
     //Find some nodejs library to do this (p-wait-for).
-    while(true){
-
-        const findAgain = await findProcess('parsecd')
-        if (findAgain.success){
-            sleep.sleep(1000,()=>{})
 
-
-        }
-        else {
-            break
-        }
-
-    }
+    await pWaitFor(async ()=>!(await findProcess("parsecd.exe")))
+    return error.SUCCESS
 }
 
-//Maybe all try catch in the main function?
 async function addAccount(nickname){
-    try {
-        //console.log(`Meo and ${__dirname}`)
-        let checkNameExists = checkNickname(nickname)
-        if (checkNameExists){
-            throw new Error("nickname already exists")
-        }
-        console.log("Adding account..")
-        //check if parsec is running
-        await stopParsec()
-        removeParsecUserFile()
-        console.log("Waiting for user to login")
-        let startOut = await startProcess()
-        if (!startOut.success){
-            throw new ProcessException(startOut.message)
-        }
-        while(true){
-            if(checkUserFile()){
-                createAccount(nickname)
-                break
-            }
-            else {
-                sleep.sleep(2000, ()=>{})
-            }
-
-        }
-        console.log("done")
-
+    let checkNameExists = checkNickname(nickname)
+    if (checkNameExists){
+        return error.NICKNAME_EXITS
+        //todo logger
     }
-    catch (error){
-        console.log(error.message)
+    //todo logger
+    console.log("Adding account..")
+    const error1 = await stopParsec()
+    if(error1){
+        return error1
     }
+    //todo logger
+    console.log("Waiting for user to login")
+    const error2 = await switchAccountData(nickname)
+    if(error2){
+        return error2
+    }
+    let startOut = await startProcess()
+    if (!startOut){
+        //todo error code
+        //throw new ProcessException(startOut.message)
```

Comment:
remove commented code, and u already did the thing in todo so u could remove that too

Reply:
done